spec.md

```diff
@@ -516,6 +541,36 @@ The type of the submodule instance `passthrough`{.firrtl} above is thus:
 { flip in : UInt<8>, out : UInt<8> }
 ```
 
+#### Instance Choices
+
+FIRRTL supports the specialization of designs by allowing instances to target a module chosen based on an `option`{.firrtl}.
+
+Example:
+``` firrtl
+circuit:
+  option Platform:
+    case FPGA:
+    case ASIC:
+
+  module InstanceChoice:
+    inst_choice clock_gate option Platform
+      when FPGA FPGATarget
+      default DefaultTarget
+
+  module DefaultTarget:
+
+  module FPGATarget:
+
+```
+
+The instance choice declaration specifies the instance name and names the option group based on which the choices are selected.
```

Comment:
Is `inst_choice` of an `extmodule` or `intmodule` allowed? Do all cases need to map to the same?

Reply:
Intmodules should not be allowed here, only full-fledged modules.